plonky2x/src/frontend/builder/mod.rs

```diff
@@ -235,6 +235,17 @@ impl<L: PlonkParameters<D>, const D: usize> CircuitBuilder<L, D> {
         }
     }
 
+    /// Returns 1 if i1 == i2 and 0 otherwise as a BoolVariable
+    pub fn is_equal<V: CircuitVariable>(&mut self, i1: V, i2: V) -> BoolVariable {
+        assert_eq!(i1.targets().len(), i2.targets().len());
```

Comment:
unecessary check

Reply:
this is assume circuitvariables are always sized, which right now they are but might not be true in the future. still removed it